unittests/runtime/Concurrent.cpp

```diff
@@ -42,3 +44,48 @@ TEST(ConcurrentReadableArrayTest, SingleThreaded) {
   add(1000000);
   check();
 }
+
+TEST(ConcurrentReadableArrayTest, MultiThreaded) {
+  const int writerCount = 16;
+  const int readerCount = 8;
+  const int insertCount = 100000;
+
+  struct Value {
+    int threadNumber;
+    int x;
+  };
+  ConcurrentReadableArray<Value> array;
+
+  auto writer = [&](int threadNumber) {
+    for (int i = 0; i < insertCount; i++)
+      array.push_back({ threadNumber, i });
```

Comment:
You're doing 1.6 million array appends in 16 simultaneous threads.  That seems sufficient to stress write/write concurrency.